src/lib/util/collapseHourlyMetrics.ts

```diff
@@ -21,31 +10,24 @@ const createMetricKey = (metric: IClientMetricsEnv): string => {
     ].join();
 };
 
-const sumYesNo = (
-    metrics: IClientMetricsEnv[],
-): Pick<IClientMetricsEnv, 'yes' | 'no'> => {
-    return metrics.reduce(
-        (acc, metric) => ({
-            yes: acc.yes + metric.yes,
-            no: acc.no + metric.no,
-        }),
-        {
-            yes: 0,
-            no: 0,
-        },
-    );
-};
-
 export const collapseHourlyMetrics = (
     metrics: IClientMetricsEnv[],
 ): IClientMetricsEnv[] => {
-    const hourlyMetrics = metrics.map((metric) => ({
-        ...metric,
-        timestamp: startOfHour(metric.timestamp),
-    }));
-
-    return groupBy(hourlyMetrics, createMetricKey).flatMap((group) => ({
-        ...group[0],
-        ...sumYesNo(group),
-    }));
+    const grouped = new Map<string, IClientMetricsEnv>();
+    metrics.forEach((metric) => {
+        const hourlyMetric = {
+            ...metric,
+            timestamp: startOfHour(metric.timestamp),
+        };
+        const key = createMetricKey(hourlyMetric);
+        if (!grouped[key]) {
+            grouped[key] = {
+                ...hourlyMetric,
+            };
+        } else {
+            grouped[key].yes = metric.yes + (grouped[key].yes || 0);
+            grouped[key].no = metric.no + (grouped[key].no || 0);
```

Comment:
Do we need the `|| 0`s here? I think the metrics objects all have yes/no counts by default.

Reply:
Honestly, I don't know. The type does not accept undefined: https://github.com/Unleash/unleash/blob/b5fead41f66b1f7da0484197b0adb710d7f45975/src/lib/types/stores/client-metrics-store-v2.ts#L10-L13 but since we're not strict with this, it could potentially be. We could consider that a bug, but I prefer to play safe. If you think it's not required and safe to be removed that'd be ok for me